src/enemy/gibdo.c

```diff
@@ -0,0 +1,434 @@
+                    this->direction = y;
+                    this->animationState = y / 8;
+                    this->field_0x74.HWORD = 300;
+                    this->field_0x78.HWORD = gUnk_020000B0->x.HALF.HI;
+                    this->field_0x7a.HWORD = gUnk_020000B0->y.HALF.HI;
+                    InitAnimationForceUpdate(this, this->animationState);
+                    return 1;
+                }
+            }
+        }
+    } else {
+        this->field_0x76.HALF.LO--;
+    }
+    return 0;
+}
+END_NONMATCH
+
+u32 sub_080378B0(Entity* this) {
+    if (this->field_0x76.HALF.HI == 0) {
+        if (sub_08049DF4(1) != 0) {
+            if (sub_0804A044(this, gUnk_020000B0, 0xa) == this->direction)
+                if (sub_0806FCB8(this, gUnk_020000B0->x.HALF.HI, gUnk_020000B0->y.HALF.HI, 0x18) != 0) {
+                    this->action = 5;
+                    this->nonPlanarMovement = 0x100;
+                    InitAnimationForceUpdate(this, this->animationState + 8);
+                    return 1;
+                }
+        }
+    } else {
+        this->field_0x76.HALF.HI--;
+    }
+    return 0;
+}
+
+u32 sub_08037914(Entity* this) {
+    if (sub_08049FDC(this, 1) != 0) {
+        if (sub_0806FCB8(gUnk_020000B0, this->field_0x78.HWORD, this->field_0x7a.HWORD, 0x28) == 0) {
+            this->field_0x78.HWORD = gUnk_020000B0->x.HALF_U.HI;
+            this->field_0x7a.HWORD = gUnk_020000B0->y.HALF_U.HI;
+            sub_0803797C(this);
+            return 0;
+        }
+        if (sub_0806FCB8(this, this->field_0x78.HWORD, this->field_0x7a.HWORD, 0x8) == 0) {
+            return 0;
+        }
+        this->field_0x76.HALF.LO = 0x3c;
+    }
+    sub_08037794(this);
+    return 1;
+}
+NONMATCH("asm/non_matching/gibdo/sub_0803797C.inc", void sub_0803797C(Entity* this)) {
+    u32 z;
+    u32 m;
+    this->actionDelay = 0x18;
+    this->field_0xf = 0x8;
+    z = sub_080045D4(this->x.HALF.HI, this->y.HALF.HI, this->field_0x78.HWORD, this->field_0x7a.HWORD);
+    m = (z + 4) & 0x18;
+    this->direction = m;
+    m = m / 8;
+    if (m != this->animationState) {
+        this->animationState = m;
+        InitAnimationForceUpdate(this, m + 4);
+    }
+}
+END_NONMATCH
+
+void sub_080379BC(Entity* this) {
+    u32 r;
+    this->actionDelay = 0x18;
+    this->field_0xf = 0x8;
+    r = Random();
+    this->animationState = ((this->animationState + (r & 2)) - 1) & 3;
+    ;
+    this->direction = this->animationState << 3;
+    InitAnimationForceUpdate(this, this->animationState + 4);
+}
+
+u32 sub_080379EC(Entity* this) {
+    if (sub_0807953C() != 0) {
+        this->actionDelay--;
+        if (this->actionDelay == 0) {
+            sub_08037A58(this);
+            return 1;
+        }
+    }
+    return 0;
+}
+
+void sub_08037A14(Entity* this) {
+    u32 x;
+    this->action = 6;
+    this->actionDelay = 0x18;
+    this->spritePriority.b0 = (this->spritePriority.b0 & (this->actionDelay - 0x20));
+    x = 3;
```

Comment:
Is this temp needed?

Reply:
It isn't I just checked sorry about that